### ISSUE TITLE
Printer Head Clean function not working (HP laserjet printer model P1102W)

### ISSUE THREAD

**Issue by ghost** [OpenPrinting/system-config-printer#50]
Hello,
the "Clean Print Heads" button in the settings window gives "Print Error" message window, and does not perform the clean function.

**Comment by RussianNeuroMancer** [comment]
hp-clean response "error: No device found that support this feature." and if added via hp-setup there is not "Clean print heads" button in printern Properties.

However, if printer is configured via CUPS web-interface to use recommended "HP LaserJet Pro P1102w Foomatic/foo2zjs-z2" driver there is "Clean print heads" button, which is assume shouldn't be here, since it not displayed with hplip driver. Is it system-config-printer bug or driver bug?

**Comment by pgmer6809** [comment]
I am also having problems with a HP 1102W. Cups with MINT19 (based on Ubuntu 18.04) will not print a test page. But Cups with Mint 18.3 (based on ubuntu 14.04) succeeds fine.
The troubleshooter gives the following info. Follwoing that I have included the result of checking the logs with journalctl
------------------------------------------------------------
Page 1 (<troubleshoot.SchedulerNotRunning.SchedulerNotRunning object at 0x7f940ca2b0f0>):
{'cups_connection_failure': False}
Page 2 (<troubleshoot.CheckLocalServerPublishing.CheckLocalServerPublishing object at 0x7f941a250a90>):
{'local_server_exporting_printers': False}
Page 3 (<troubleshoot.ChoosePrinter.ChoosePrinter object at 0x7f941a250fd0>):
{'cups_dests_available': [], 'cups_queue_listed': False}
Page 4 (<troubleshoot.LocalOrRemote.LocalOrRemote object at 0x7f941a1f81d0>):
{'printer_is_remote': True}
Page 5 (<troubleshoot.RemoteAddress.RemoteAddress object at 0x7f941413d6d8>):
{'remote_server_ip_address': '192.168.1.65:9100',
 'remote_server_name': 'NPI869D52'}
Page 6 (<troubleshoot.CheckNetworkServerSanity.CheckNetworkServerSanity object at 0x7f941a202a90>):
{'remote_server_connect_ipp': False,
 'remote_server_name_resolves': False,
 'remote_server_traceroute': ([''], ['traceroute: unknown host', ''], 1),
 'remote_server_try_connect': '192.168.1.65:9100'}
Page 7 (<troubleshoot.Locale.Locale object at 0x7f941a202828>):
{'printer_page_size': None,
 'system_locale_lang': None,
 'user_locale_ctype': 'en_CA',
 'user_locale_messages': 'en_CA'}
----------------------------------------------------------
=======================================
- Logs begin at Tue 2018-09-04 06:40:43 PDT, end at Fri 2018-09-07 23:36:04 PDT. --
Sep 07 07:47:38 trojan systemd[1]: Started CUPS Scheduler.
Sep 07 07:49:30 trojan systemd[1]: Stopping CUPS Scheduler...
Sep 07 07:49:31 trojan systemd[1]: Stopped CUPS Scheduler.
-- Reboot --
Sep 07 08:17:56 trojan systemd[1]: Started CUPS Scheduler.
Sep 07 08:22:56 trojan systemd[1]: Stopping CUPS Scheduler...
Sep 07 08:22:56 trojan systemd[1]: Stopped CUPS Scheduler.
Sep 07 08:22:56 trojan systemd[1]: Started CUPS Scheduler.
Sep 07 12:12:51 trojan systemd[1]: Stopping CUPS Scheduler...
Sep 07 12:12:52 trojan systemd[1]: Stopped CUPS Scheduler.
-- Reboot --
Sep 07 19:43:28 trojan systemd[1]: Started CUPS Scheduler.
Sep 07 19:45:15 trojan systemd[1]: Stopping CUPS Scheduler...
Sep 07 19:45:16 trojan systemd[1]: Stopped CUPS Scheduler.
-- Reboot --
Sep 07 22:36:22 trojan systemd[1]: Started CUPS Scheduler.
Sep 07 22:37:32 trojan /hpfax[2115]: [2115]: error: Failed to create /var/spool/cups/tmp/.hplip
Sep 07 22:38:27 trojan foo2zjs-wrapper[2214]: foo2zjs -r1200x600 -g10200x6600 -p1 -m1 -n1 -d1 -s7 -z2  -u 192x96 -l 192x96 -L 0  -T3    -P
Sep 07 22:40:03 trojan systemd[1]: cups.service: Main process exited, code=exited, status=1/FAILURE
Sep 07 22:40:03 trojan systemd[1]: cups.service: Failed with result 'exit-code'.
Sep 07 22:40:03 trojan systemd[1]: cups.service: Service hold-off time over, scheduling restart.
Sep 07 22:40:03 trojan systemd[1]: cups.service: Scheduled restart job, restart counter is at 1.
Sep 07 22:40:20 trojan systemd[1]: Stopped CUPS Scheduler.
Sep 07 22:40:20 trojan systemd[1]: Started CUPS Scheduler.
Sep 07 22:46:52 trojan foo2zjs-wrapper[2447]: foo2zjs -r1200x600 -g10200x6600 -p1 -m1 -n1 -d1 -s7 -z2  -u 192x96 -l 192x96 -L 0  -T3    -P
Sep 07 22:52:17 trojan /hpfax[3003]: [3003]: error: Failed to create /var/spool/cups/tmp/.hplip
Sep 07 23:16:21 trojan systemd[1]: cups.service: Main process exited, code=exited, status=1/FAILURE
Sep 07 23:16:21 trojan systemd[1]: cups.service: Failed with result 'exit-code'.
Sep 07 23:16:21 trojan systemd[1]: cups.service: Service hold-off time over, scheduling restart.
Sep 07 23:16:21 trojan systemd[1]: cups.service: Scheduled restart job, restart counter is at 2.
Sep 07 23:16:38 trojan systemd[1]: Stopped CUPS Scheduler.
Sep 07 23:16:38 trojan systemd[1]: Started CUPS Scheduler.
===================================================

**Comment by pgmer6809** [comment]
Sorry for the bold face in previous comment. Don't know how that happened

**Comment by zdohnal** [comment]
@pgmer6809 Hi,

it seems like CUPS or foo2zjs error, please report there.

**Comment by tillkamppeter** [comment]
@pgmer6809, I have fixed your post now. Please use proper markdown and not ASCII-art separator lines.

**Comment by tillkamppeter** [comment]
@pgmer6809, please note that your printer is a laser printer and laser printers do not have head cleaning functionality.
@zdohnal, if system-config-printer displays a "Clean heads" button with this printer, this is a bug, if it does not display such a button, all is as it should be.

**Comment by zdohnal** [comment]
@tillkamppeter s-c-p decides whether display the button or not according cupsd get-printer-attributes response contents.

This is CUPS bug - cupsd returns printer-commands attribute in get-printer-attributes response and it returns all 3 values for the uri used in cups temporary queue functionality or CUPS server uri (ipp://<server>:631/printers/<queue>), or 'none' for basic ipp uris (ipp://<printer_ip>:631/ipp/print).

I tested both scenarios with laser printers which I have access to.

I reported it to CUPS as https://github.com/apple/cups/issues/5780 .
